functions/src/open-api-spec-patcher.ts

```diff
@@ -0,0 +1,56 @@
+export class OpenApiSpecPatcher {
+  options: any;
+
+  constructor(options?: any) {
+    this.options = { pathsToRemove: [], patchServerSlashPrefixes: true, ...(options || {}) };
+  }
+
+  public patchJson(spec: string): string {
+    const obj = JSON.parse(spec);
```

Comment:
Do you refer to making shallow copies etc?

Reply:
I was just thinking on something like this: 

```typescript
const obj = JSON.parse(spec, (key, value) => { if (key == 'some-key') return 'new-value' });
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#parameters

I think it's either overkill or not working, because someone needs access to another part of the spec...